### PR TITLE
spec file: Disable OCR support on SLE and fix condition for Python support

### PR DESCRIPTION
**Pull request by Martchus**
See particular commit messages for details and https://progress.opensuse.org/issues/128318

### CODE REVIEW

**Review by kalikiana — APPROVED** [review]

**Comment by perlpunk** [comment]
```
[   31s] error: parse error in expression:  0%{ocr_not_supported_by_arch} || !01
[   31s] error:                              ^
[   31s] error: /home/abuild/rpmbuild/SOURCES/os-autoinst.spec:91: bad %if condition:  0%{ocr_not_supported_by_arch} || !01
```

File: dist/rpm/os-autoinst.spec [hunk 1]
```diff
@@ -68,7 +68,7 @@ Source0:        %{name}-%{version}.tar.xz
 %bcond_with black
 %endif
 # SLE is missing Python support requirements
-%if !0%{?is_opensuse}
+%if 0%{?is_opensuse}
```
Comment (kalikiana) [inline dist/rpm/os-autoinst.spec:71]: Ah. I guess you need to get rid of the ? here 🤔 
```suggestion
%if 0%{is_opensuse}
```

Comment (baierjan) [reply dist/rpm/os-autoinst.spec:71]: Nope `0%{?is_opensuse}` is correct syntax: https://en.opensuse.org/openSUSE:Specfile_guidelines#Conditionals

File: dist/rpm/os-autoinst.spec [hunk 2]
```diff
@@ -85,10 +85,13 @@ Source0:        %{name}-%{version}.tar.xz
 %else
 %define python_style_requires %{nil}
 %endif
-%ifnarch ppc ppc64 ppc64le s390x
-%bcond_without ocr
-%else
+%ifarch ppc ppc64 ppc64le s390x
+%define ocr_not_supported_by_arch 1
+%endif
+%if 0%{ocr_not_supported_by_arch} || !0%{?is_opensuse}
```
Comment (baierjan) [inline dist/rpm/os-autoinst.spec:91]: From https://rpm-software-management.github.io/rpm/manual/macros.html

```suggestion
%ifarch ppc ppc64 ppc64le s390x
%define ocr_not_supported_by_arch 1
%endif
%if 0%{?ocr_not_supported_by_arch} || !0%{?is_opensuse}
```

Or, looking at https://github.com/rpm-software-management/rpm/blob/master/docs/manual/conditionalbuilds.md#creating-build-conditionals you might probably use something like 

```suggestion
%ifarch ppc ppc64 ppc64le s390x
%bcond ocr_not_supported_by_arch 1
%endif
%if %{with ocr_not_supported_by_arch} || !0%{?is_opensuse}
```

_Note: this is syntax suggestion only, I did not try to understand the full logic workflow to optimize the conditionals._

**Pull request merged**